src/core/structured_config.py

```diff
@@ -264,3 +266,12 @@ def secret_validator(cls, value: str) -> str:
             )
 
         return value
+
+    @field_validator("roles_mapping", mode="before")
+    @classmethod
+    def parse_roles_mapping_to_dict(cls, v: str) -> dict:
+        """Validator to convert a string representation of a dict into a dict."""
+        try:
+            return json.loads(v)
+        except json.JSONDecodeError as e:
+            raise ValueError("roles-mapping is not a valid JSON dictionary") from e
```

Comment:
I'm realizing that these id will directly go into `--allow-principal=User:{username}`. Should we sanitize it a bit more, to make sure values on the dict are uuid based. Maybe not all users in the mapping are uuid based, but at least remove spaces or similar.